### PR TITLE
separate inputs for encoding and decoding spikes

### PR DESCRIPTION
**Pull request by droumis**
along with some smaller changes, including how get_irregular_resampled is done

### CODE REVIEW

**Comment by daliu87** [comment]
Why did you delete `get_mapped_single_axis()` from `LinearPosition`?  That's the transformation command to convert a current `LinearPosition` into a new `FlatLinearPosition` using the existing data in `LinearPositiion`.  Right now there doesn't appear to be any way to do the transformation on existing data into a new object.

**Comment by droumis** [comment]
hmm, great question! 
I had been creating instances of `FlatLinearPosition` directly from a df created from eric's code, not through a `LinearPosition` object. Also, I think there was something funky with the original function, so I rewrote it and put the new version as a method of `FlatLinearPosition`, and then decided to just delete it from `LinearPosition`. THAT SAID, I think you're right and I can fix this

File: spykshrk/franklab/data_containers.py [hunk 1]
```diff
@@ -575,7 +575,7 @@ def create_default(cls, df, sampling_rate, arm_coord, parent=None, **kwds):
         if parent is None:
             parent = df
 
-        return cls(df=df, sampling_rate=sampling_rate, arm_coord=arm_coord, parent=parent, **kwds)
+        return cls(data=df, sampling_rate=sampling_rate, arm_coord=arm_coord, parent=parent, **kwds)
```
Comment (droumis) [inline spykshrk/franklab/data_containers.py:578]: i changed this because the super (DayEpochTimeseries) __init__ looks for 'data' kwarg

**Comment by daliu87** [comment]
Looking more closely at your `get_mapped_single_axis()` it doesn't make sense to me, your position format must be pretty different.  My `LinearPosition` really should be more specifically `FilterFrameworkLinearPosition` and there should be a new class for your format.  They should both still create `FlatLinearPosition`.

For now I'm going to rename and make private your function and restore my original function (so you can still use your function with a rename if you pull from my repo).  We'll figure out the proper way to organize this after resolving this pull request.

**Comment by droumis** [comment]
ok that works too. i think we should converge at some point, but not at the expense of moving forward right now. thanks

**Comment by daliu87** [comment]
The changes you made to `get_irregular_resampled()` has a couple of issues.  It's a member of `DayEpochTimeSeries` so it should not be hard coded to return specific subclasses (like `FlatLinearPosition`), instead the class it returns should be the same as `type(self)`.

I'm also getting a warning:
`UserWarning: merging between different levels can give an unintended result (1 levels on the left, 2 on the right)`

I'll look into an alternative way to doing this, my gut feeling is `pd.merge_asof()' isn't the exact method we want to use.

**Pull request merged**